src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -489,5 +496,14 @@ protected override int QuickSelect(int[] map, int right, int idx)
 
             return map[idx];
         }
+
+        protected override int Min(int[] map, int count)
+        {
+            int index = 0;
+            for (int i = 1; i < count; i++)
+                if (CompareKeys(map[i], map[index]) < 0)
```

Comment:
If two elements are equal according to CompareKeys, does this guarantee to always return the same element that OrderBy(...).Take(1) previously would?

Reply:
Internally, `CompareKeys` returns `index1 - index2` for equal objects to ensure the sorting stability. Thus, we will always be consistent with the old behavior.